source/Fleet.cpp

```diff
@@ -96,30 +96,35 @@ void Fleet::Enter(const System &system, list<shared_ptr<Ship>> &ships, const Pla
 {
 	if(!total || !government || variants.empty())
 		return;
-	
+
 	// Pick a fleet variant to instantiate.
 	const Variant &variant = ChooseVariant();
 	if(variant.ships.empty())
 		return;
-	
+
 	// Where this fleet can come from depends on whether it is friendly to any
 	// planets in this system and whether it has jump drives.
 	vector<const System *> linkVector;
 	// Find out what the "best" jump method the fleet has is. Assume that if the
 	// others don't have that jump method, they are being carried as fighters.
 	// That is, content creators should avoid creating fleets with a mix of jump
 	// drives and hyperdrives.
-	int jumpType = 0;
+	bool hasJumpDrive = false;
+	double jumpType = 0.;
```

Comment:
for consistency, I would ask you to go ahead and change (rename) this then. jumpType is then no longer a "Type" but a "Cost" variable.
In Ship.h you defined a new enum JumpType for that purpose.
Now you have two pretty different types of "jumpType".
In this case, you only need the ability and do not need the cost at all. The entire Fleet.cpp contained only 5x the word "jumpType". So, if you change it to hasJumpDrive, you dont neven need your double jumpType (with the meaning of jumpCost) anymore.

Also, you could explicitely overcome the original comment in the lines above and provide a solution which handles "mixed" fleets.
I thought about doing that but will not do it right now in order to avoid code conflicts.
My approach would be to take `jumpType = min(jumpType, ship->GetJumpType())` for all ships which can't be carried. Otherwise the fighters and drones, which have jumpType=0 (or NONE), would mess up because of the min-operator.